segmentation/include/pcl/segmentation/euclidean_cluster_comparator.h

```diff
@@ -48,225 +48,141 @@
-          return (dist < dist_threshold);
-        }
-
-      protected:
-
-
-        /** \brief Set of labels with similar size as the input point cloud,
-          * aggregating points into groups based on a similar label identifier.
-          *
-          * It needs to be set in conjunction with the \ref exclude_labels_
-          * member in order to provided a masking functionality.
-          */
-        PointCloudLPtr labels_;
-
-        /** \brief Specifies which labels should be excluded com being clustered.
-          *
-          * If a label is not specified, it's assumed by default that it's
-          * intended be excluded
-          */
-        ExcludeLabelSetConstPtr exclude_labels_;
-
-        float distance_threshold_;
-
-        bool depth_dependent_;
-
-        Eigen::Vector3f z_axis_;
-    };
-  } // namespace experimental
-
-
-  /** \brief EuclideanClusterComparator is a comparator used for finding clusters based on euclidian distance.
-    *
-    * \author Alex Trevor
-    */
-  template<typename PointT, typename PointNT, typename PointLT = deprecated::T>
-  class EuclideanClusterComparator : public experimental::EuclideanClusterComparator<PointT, PointLT>
+  template<typename PointT, typename PointLT = pcl::Label>
+  class EuclideanClusterComparator : public ::pcl::Comparator<PointT>
   {
     protected:
 
-      using experimental::EuclideanClusterComparator<PointT, PointLT>::exclude_labels_;
+      using pcl::Comparator<PointT>::input_;
 
     public:
+      using typename Comparator<PointT>::PointCloud;
+      using typename Comparator<PointT>::PointCloudConstPtr;
 
-      using PointCloudN = pcl::PointCloud<PointNT>;
-      using PointCloudNPtr = typename PointCloudN::Ptr;
-      using PointCloudNConstPtr = typename PointCloudN::ConstPtr;
+      using PointCloudL = pcl::PointCloud<PointLT>;
+      using PointCloudLPtr = typename PointCloudL::Ptr;
+      using PointCloudLConstPtr = typename PointCloudL::ConstPtr;
 
-      using Ptr = shared_ptr<EuclideanClusterComparator<PointT, PointNT, PointLT> >;
-      using ConstPtr = shared_ptr<const EuclideanClusterComparator<PointT, PointNT, PointLT> >;
+      using Ptr = shared_ptr<EuclideanClusterComparator<PointT, PointLT> >;
+      using ConstPtr = shared_ptr<const EuclideanClusterComparator<PointT, PointLT> >;
 
-      using experimental::EuclideanClusterComparator<PointT, PointLT>::setExcludeLabels;
+      using ExcludeLabelSet = std::set<std::uint32_t>;
+      using ExcludeLabelSetPtr = shared_ptr<ExcludeLabelSet>;
+      using ExcludeLabelSetConstPtr = shared_ptr<const ExcludeLabelSet>;
 
       /** \brief Default constructor for EuclideanClusterComparator. */
-      PCL_DEPRECATED(1, 12, "remove PointNT from template parameters")
       EuclideanClusterComparator ()
-        : normals_ ()
-        , angular_threshold_ (0.0f)
+        : distance_threshold_ (0.005f)
+        , depth_dependent_ ()
       {}
```

Comment:
Replace with `EuclideanClusterComparator() = default;` and use default init for the variables?